server/ladders-remote.js

```diff
@@ -95,7 +95,7 @@ class LadderStore {
 			Monitor.warn(`room expired before ladder update was received`);
 			return [p1score, null, null];
 		}
-		if (data.errorip) {
+		if (data && data.errorip) {
```

Comment:
the Promise may be null, so we have to check for that

Reply:
I'd check for it by `if (!data) return [p1score, null, null];` with an error message about a failed request.

But also TypeScript has never complained about this, and also this code hasn't encountered crashes in the last very many years, so maybe you're wrong about this being possibly null?

Reply:
TypeScript has never complained about this because it never had typing information for LoginServer and subsequently any of its methods. I didn't say it will possibly be null, but API wise it can. If you remove the checks, typescript will error this.
![image](https://user-images.githubusercontent.com/7896469/52240420-12c9c600-28d1-11e9-8dfb-70b40071adcd.png)